File: Dopamine.CollectionModule/ViewModels/CollectionSubMenuViewModel.vb
Imports Dopamine.Core.Settings
Imports Microsoft.Practices.Prism.Mvvm

Namespace ViewModels
    Public Class CollectionSubMenuViewModel
        Inherits BindableBase

#Region "Variables"
        Private mIsArtistsSelected As Boolean
        Private mIsGenresSelected As Boolean
        Private mIsAlbumsSelected As Boolean
        Private mIsTracksSelected As Boolean
        Private mIsPlaylistsSelected As Boolean
        Private mIsCloudSelected As Boolean
#End Region

#Region "Properties"
        Public Property IsArtistsSelected() As Boolean
            Get
                Return mIsArtistsSelected
            End Get
            Set(ByVal value As Boolean)
                SetProperty(Of Boolean)(mIsArtistsSelected, value)
            End Set
        End Property

        Public Property IsGenresSelected() As Boolean
            Get
                Return mIsGenresSelected
            End Get
            Set(ByVal value As Boolean)
                SetProperty(Of Boolean)(mIsGenresSelected, value)
            End Set
        End Property

        Public Property IsAlbumsSelected() As Boolean
            Get
                Return mIsAlbumsSelected
            End Get
            Set(ByVal value As Boolean)
                SetProperty(Of Boolean)(mIsAlbumsSelected, value)
            End Set
        End Property

        Public Property IsTracksSelected() As Boolean
            Get
                Return mIsTracksSelected
            End Get
            Set(ByVal value As Boolean)
                SetProperty(Of Boolean)(mIsTracksSelected, value)
            End Set
        End Property

        Public Property IsPlaylistsSelected() As Boolean
            Get
                Return mIsPlaylistsSelected
            End Get
            Set(ByVal value As Boolean)
                SetProperty(Of Boolean)(mIsPlaylistsSelected, value)
            End Set
        End Property

        Public Property IsCloudSelected() As Boolean
            Get
                Return mIsCloudSelected
            End Get
            Set(ByVal value As Boolean)
                SetProperty(Of Boolean)(mIsCloudSelected, value)
            End Set
        End Property
#End Region

#Region "Construction"
        Public Sub New()

            Me.SelectMenuItem()
        End Sub
#End Region

#Region "Private"

        Private Sub SelectMenuItem()

            If XmlSettingsClient.Instance.Get(Of Boolean)("Startup", "ShowLastSelectedPage") Then
                Dim screen As SelectedPage = CType(XmlSettingsClient.Instance.Get(Of Integer)("FullPlayer", "SelectedPage"), SelectedPage)

                Select Case screen
                    Case SelectedPage.Artists
                        Me.IsArtistsSelected = True
                    Case SelectedPage.Genres
                        Me.IsGenresSelected = True
                    Case SelectedPage.Albums
                        Me.IsAlbumsSelected = True
                    Case SelectedPage.Tracks
                        Me.IsTracksSelected = True
                    Case SelectedPage.Playlists
                        Me.IsPlaylistsSelected = True
                    Case SelectedPage.Recent
                        Me.IsCloudSelected = True
                    Case Else
                        Me.IsArtistsSelected = True
                End Select
            Else
                Me.IsArtistsSelected = True
            End If
        End Sub
#End Region
    End Class
End Namespace

File: Dopamine.CollectionModule/ViewModels/CollectionSubMenuViewModel.cs
using Dopamine.Core.Settings;
using Prism.Mvvm;

namespace Dopamine.CollectionModule.ViewModels
{
    public class CollectionSubMenuViewModel : BindableBase
    {
        #region Variables
        private bool isArtistsSelected;
        private bool isGenresSelected;
        private bool isAlbumsSelected;
        private bool isTracksSelected;
        private bool isPlaylistsSelected;
        private bool isCloudSelected;
        #endregion

        #region Properties
        public bool IsArtistsSelected
        {
            get { return this.isArtistsSelected; }
            set { SetProperty<bool>(ref this.isArtistsSelected, value); }
        }

        public bool IsGenresSelected
        {
            get { return this.isGenresSelected; }
            set { SetProperty<bool>(ref this.isGenresSelected, value); }
        }

        public bool IsAlbumsSelected
        {
            get { return this.isAlbumsSelected; }
            set { SetProperty<bool>(ref this.isAlbumsSelected, value); }
        }

        public bool IsTracksSelected
        {
            get { return this.isTracksSelected; }
            set { SetProperty<bool>(ref this.isTracksSelected, value); }
        }

        public bool IsPlaylistsSelected
        {
            get { return this.isPlaylistsSelected; }
            set { SetProperty<bool>(ref this.isPlaylistsSelected, value); }
        }

        public bool IsCloudSelected
        {
            get { return this.isCloudSelected; }
            set { SetProperty<bool>(ref this.isCloudSelected, value); }
        }
        #endregion

        #region Construction
        public CollectionSubMenuViewModel()
        {
            this.SelectMenuItem();
        }
        #endregion

        #region Private
        private void SelectMenuItem()
        {
            if (XmlSettingsClient.Instance.Get<bool>("Startup", "ShowLastSelectedPage"))
            {
                SelectedPage screen = (SelectedPage)XmlSettingsClient.Instance.Get<int>("FullPlayer", "SelectedPage");

                switch (screen)
                {
                    case SelectedPage.Artists:
                        this.IsArtistsSelected = true;
                        break;
                    case SelectedPage.Genres:
                        this.IsGenresSelected = true;
                        break;
                    case SelectedPage.Albums:
                        this.IsAlbumsSelected = true;
                        break;
                    case SelectedPage.Tracks:
                        this.IsTracksSelected = true;
                        break;
                    case SelectedPage.Playlists:
                        this.IsPlaylistsSelected = true;
                        break;
                    case SelectedPage.Recent:
                        this.IsCloudSelected = true;
                        break;
                    default:
                        this.IsArtistsSelected = true;
                        break;
                }
            }
            else
            {
                this.IsArtistsSelected = true;
            }
        }
        #endregion
    }
}
